lib/index.js

```diff
@@ -52,37 +52,37 @@ class PackageJson {
     'binRefs',
   ])
 
-  // default behavior, just loads and parses
+  // default behavior, loads a package.json at given path and JSON parses
   static async load (path) {
-    return await new PackageJson(path).load()
+    const p = new PackageJson()
+    return p.load(path)
   }
 
   // read-package-json compatible behavior
   static async prepare (path, opts) {
-    return await new PackageJson(path).prepare(opts)
+    const p = new PackageJson()
+    await p.load(path, true)
+    return p.prepare(opts)
   }
 
   // read-package-json-fast compatible behavior
   static async normalize (path, opts) {
-    return await new PackageJson(path).normalize(opts)
+    const p = new PackageJson()
+    await p.load(path)
+    return p.normalize(opts)
   }
 
-  #filename
   #path
-  #manifest = {}
+  #manifest
   #readFileContent = ''
-  #fromIndex = false
+  #canSave = true
 
-  constructor (path) {
+  // Load content from given path
+  async load (path, parseIndex) {
```

Comment:
It's still a breaking change since this method signature changes, but would it make refactoring easier if we still allowed it in the constructor?

I also think it's a valid use case to create an instance when you know the path and then call load on it at a later time.

```js
{
  constructor(path) {
      this.#path = path
  }
  async load (path = this.#path, parseIndex) {

  }
}
```

Reply:
We don't currently have that use case, and we are trying to fully decouple the "read a file" from everything else this module does. Eventually it needs to read raw JSON from the packument in the registry.  Having a dedicated "lane" where we know we have read from a file or not helps out immensely in reducing complexity